### DISCUSSION TITLE
Mesh fracturing to smaller pieces

### DISCUSSION THREAD

**Discussion by ttsesm** [pyvista/pyvista#2291] [Q&A]
Hi guys,

Is it possible to use pyvista to apply some 3d mesh cell fracturing to smaller pieces? Something like that https://www.youtube.com/watch?v=ZG_ZMnKzVTQ 

I found the following corresponding issue https://github.com/pyvista/pyvista-support/issues/144 but the fracturing seems to be done as slices rather than randomly. I would prefer to be in a random way similarly to what is shown in the linked video above.

Thanks.

**Comment by RichardScottOZ** [comment]
You want to select random, but all different pieces?

**Reply by ttsesm** [reply]
Yes, imagine you have a solid (or non solid) sphere (this could be any 3D mesh) and then I want to fracture this in random pieces where each piece is fractured differently from the other. The when you put the pieces back together you should get the original sphere.

The fracturing though I would prefer not to be in a sharp cut like a slice. It might contain such fractures but not only such, I would like to have something similar like when you break an object and is exemplified in the blender addon in the video.

**Reply by RichardScottOZ** [reply]
Do you have the random piece generator already?

**Reply by ttsesm** [reply]
Actually I got something running by using the `vedo` lib, the `pyvista_tetget` and `meshfix`, you can have a look [here](https://github.com/marcomusy/vedo/issues/498#issuecomment-1061841721). The problem now is how to optimize the tetrahedralization so that I do not have such rough cuts in the individual pieces.

**Comment by RichardScottOZ** [comment]
Nice - 'how to optimally divide a chocolate easter rabbit' would be a fun post for next month.
